dallinger/deployment.py

```diff
@@ -237,6 +237,31 @@ def assemble_experiment_temp_dir(config):
         src = os.path.join(dallinger_root, "heroku", "Procfile_no_clock")
         shutil.copy(src, os.path.join(dst, "Procfile"))
 
+    from dallinger.config import initialize_experiment_package
+
+    initialize_experiment_package(dst)
+    from dallinger.experiment import load
+
+    exp_class = load()
+    extra_files = getattr(exp_class, "extra_files", None)
+    if extra_files is None:
+        try:
+            from dallinger_experiment.experiment import extra_files
+        except ImportError:
+            try:
+                from dallinger_experiment.dallinger_experiment import extra_files
+            except ImportError:
+                pass
+
+    if extra_files is not None:
+        for src, filename in extra_files():
+            filename = filename.lstrip("/")
+            dst_filepath = os.path.join(dst, filename)
+            if os.path.isdir(src):
+                shutil.copytree(src, dst_filepath)
+            else:
+                shutil.copyfile(src, dst_filepath)
+
```

Comment:
We'll now have a second source of experiment files that's not included in the size checks performed by `ExperimentFileSource`, unfortunately. I wonder if that class should include this new functionality, so we'd just have one implementation of the concept?

Reply:
The problem here is that there was a specific requirement for them to be included after the files from Dallinger, which are done after the `ExperimentFileSource` is parsed. I could definitely be convinced to have `ExperimentFileSource`, `DallingerFileSource` and `DynamicFileSource`.

Reply:
That makes sense... too bad. Could we extract `DynamicFileSource` for now without too much trouble? I'd already thought of the helpful conceptual parallelism of a `DallingerFileSource` when I put together `ExperimentFileSource`, and I think if we now had 2 out of 3, it would push us to do the 3rd at some point. There's so much detail in deployment; I think pulling up some higher-level concepts really helps.

Reply:
Agreed! I'll do that.